src/main/app/pay/payClient.ts

```diff
@@ -151,10 +95,41 @@ export class GovPayClient implements PayClient {
     }
   }
 
-  private preparePaymentRequest (caseReference: string, externalId: string, fees: Fee[]): object {
+  async update (user: User, paymentReference: string, caseReference: string, caseNumber: string): Promise<void> {
+    checkDefined(user, 'User is required')
+    checkNotEmpty(paymentReference, 'Payment reference is required')
+    checkNotEmpty(caseReference, 'Case Reference is required')
+    checkNotEmpty(caseNumber, 'Case Number is required')
+    try {
+      await request.patch({
+        uri: `${paymentURL}/${paymentReference}`,
+        body: this.preparePaymentUpdateRequest(caseReference, caseNumber),
+        headers: {
+          Authorization: `Bearer ${user.bearerToken}`,
+          ServiceAuthorization: `Bearer ${this.serviceAuthToken.bearerToken}`
+        }
+      })
+      return
+    } catch (err) {
+      if (err.statusCode === HttpStatus.NOT_FOUND) {
+        return undefined
+      }
+      throw err
+    }
+    return Promise.resolve()
```

Comment:
This line is unreachable.
This method should return a mutated invocation of the `patch` Promise.
```
return request.patch(...)
  .success(onFulfilment => Promise.resolve())
  .catch(onRejection => Promise.reject())
```
This also eliminates the need for the `try...catch` block.

Reply:
yeah, it works. So I have made changes as suggested above. @andrewwa-kainos